crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java

```diff
@@ -172,38 +176,45 @@ public void determineRepTypeByDecisionReason(CrownCourtActionsRequestDTO request
 
     public ApiCrownCourtSummary determineRepOrderDate(CrownCourtActionsRequestDTO requestDTO) {
         ApiCrownCourtSummary crownCourtSummary = requestDTO.getCrownCourtSummary();
-        if (crownCourtSummary.getRepOrderDecision() != null && crownCourtSummary.getRepOrderDate() == null) {
+        String repOrderDecision = crownCourtSummary.getRepOrderDecision();
+        if (repOrderDecision != null && crownCourtSummary.getRepOrderDate() == null) {
             switch (requestDTO.getCaseType()) {
-                case INDICTABLE:
-                    crownCourtSummary.setRepOrderDate(requestDTO.getDecisionDate());
-                    break;
-                case EITHER_WAY:
-                    if (requestDTO.getMagCourtOutcome() == MagCourtOutcome.COMMITTED_FOR_TRIAL) {
-                        switch (requestDTO.getDecisionReason()) {
-                            case GRANTED -> crownCourtSummary.setRepOrderDate(requestDTO.getDecisionDate());
-                            case FAILIOJ, FAILMEIOJ, FAILMEANS -> crownCourtSummary.setRepOrderDate(requestDTO.getCommittalDate());
-                            default -> crownCourtSummary.setRepOrderDate(null);
-                        }
-                    }
-                    break;
-                case CC_ALREADY, COMMITAL:
-                    crownCourtSummary.setRepOrderDate(requestDTO.getDateReceived());
-                    break;
-                case APPEAL_CC:
-                    try {
-                        IOJAppealDTO iojAppealDTO = maatCourtDataService
-                                .getCurrentPassedIOJAppealFromRepId(requestDTO.getRepId(), requestDTO.getLaaTransactionId());
+                case INDICTABLE -> crownCourtSummary.setRepOrderDate(requestDTO.getDecisionDate());
+                case EITHER_WAY -> crownCourtSummary.setRepOrderDate(
+                        determineMagsRepOrderDate(requestDTO, repOrderDecision)
+                );
+                case CC_ALREADY, COMMITAL -> crownCourtSummary.setRepOrderDate(requestDTO.getDateReceived());
+                case APPEAL_CC -> {
+                    IOJAppealDTO iojAppealDTO = maatCourtDataService
+                            .getIOJAppeal(requestDTO.getRepId(), requestDTO.getLaaTransactionId());
+                    if (iojAppealDTO != null) {
```

Comment:
With this implementation you need to update the client to not throw an Exception

Reply:
I'm not sure if that's really required, I guess you'd just end up with the generic "Call to Court Data API failed" if anything other than a not found is returned.

I think this is something we need to discuss in more detail. Everything has been copied over from CMA so if we make enhancements here we need to make sure they're back ported.